src/komodo_bitcoind.cpp

```diff
@@ -2450,35 +2450,36 @@ int64_t komodo_coinsupply(int64_t *zfundsp,int64_t *sproutfundsp,int32_t height)
     return(supply);
 }
 
-struct komodo_staking *komodo_addutxo(struct komodo_staking *array,int32_t *numkp,int32_t *maxkp,uint32_t txtime,uint64_t nValue,uint256 txid,int32_t vout,char *address,uint8_t *hashbuf,CScript pk)
+void komodo_addutxo(std::vector<komodo_staking> &array,uint32_t txtime,uint64_t nValue,uint256 txid,int32_t vout,char *address,uint8_t *hashbuf,CScript pk)
 {
-    uint256 hash; uint32_t segid32; struct komodo_staking *kp;
+    uint256 hash; uint32_t segid32; komodo_staking kp;
     segid32 = komodo_stakehash(&hash,address,hashbuf,txid,vout);
-    if ( *numkp >= *maxkp )
-    {
-        *maxkp += 1000;
-        array = (struct komodo_staking *)realloc(array,sizeof(*array) * (*maxkp));
-        //fprintf(stderr,"realloc max.%d array.%p\n",*maxkp,array);
-    }
-    kp = &array[(*numkp)++];
-    //fprintf(stderr,"kp.%p num.%d\n",kp,*numkp);
-    memset(kp,0,sizeof(*kp));
-    strcpy(kp->address,address);
-    kp->txid = txid;
-    kp->vout = vout;
-    kp->hashval = UintToArith256(hash);
-    kp->txtime = txtime;
-    kp->segid32 = segid32;
-    kp->nValue = nValue;
-    kp->scriptPubKey = pk;
-    return(array);
+    if ( array.size() >= array.capacity() )
+    {
+        array.reserve(array.capacity() + 1000);
+        //fprintf(stderr,"%s realloc array.size().%d array.capacity().%d\n", __func__, array.size(), array.capacity());
+    }
+    //memset(&kp,0,sizeof(kp));
+    strcpy(kp.address, address);
+    kp.txid = txid;
+    kp.vout = vout;
+    kp.hashval = UintToArith256(hash);
+    kp.txtime = txtime;
+    kp.segid32 = segid32;
+    kp.nValue = nValue;
+    kp.scriptPubKey = pk;
+    array.push_back(kp);
+    //fprintf(stderr,"kp.%p array.size().%d\n",kp,array.size());
 }
 
 int32_t komodo_staked(CMutableTransaction &txNew,uint32_t nBits,uint32_t *blocktimep,uint32_t *txtimep,uint256 *utxotxidp,int32_t *utxovoutp,uint64_t *utxovaluep,uint8_t *utxosig, uint256 merkleroot)
 {
-    static struct komodo_staking *array; static int32_t numkp,maxkp; static uint32_t lasttime;
+    // use thread_local to prevent crash in case of accidental thread overlapping
+    thread_local std::vector<komodo_staking> array; 
```

Comment:
thread_local here does not automatically make it static. See https://stackoverflow.com/questions/22794382/are-c11-thread-local-variables-automatically-static